docs/unidling.md

```diff
@@ -0,0 +1,188 @@
+# Unidling Pods
+## Abstract
+
+Unidling is an event-driven feature that allows users to automatically create destination pods for
+an idled service when a connection is made to it.
+
+## Motivation
+
+Not all pods in a Kubernetes cluster will be active at all times.  When all of the pods
+resolving to a service are inactive, new ones must be created when a new request for that service
+is received.  The process of creating new pods for an idled service in response to a new request to
+that service is called "unidling".  This proposal covers how unidling should be performed in
+Kubernetes.
+
+We refer to the process of determining which pods to stop and stopping them as "idling"; it
+will be dealt with in a separate proposal and is out of scope for this proposal.  
+
+## Lexicon
+
+- **Destination pods**: the backend pods (or other backends as necessary) that a service points to
+- **Idled service**: a service with zero endpoints
+
+## Constraints and Assumptions
+
+- Unidling should be orthogonal to autoscaling and idling
+- No specific idling mechanism or autoscaler is assumed
+- The case where a service is backed by pods which are not managed by a replication controller: 
+  - Not covered by this proposal at all
+  - Requests to unidle these service should be gracefully ignored by components scoped to this
+    proposal
+  - A separate unidler may be written for these
+
+## Use Cases
+
+The following use cases are explored by this proposal:
+
+1.  As a cluster operator, I want new request to an idled service to trigger the creation of new
+    destination pods for that service
+2.  As a cluster operator, I want requests to an idled service to be buffered while the service is
+    being unidled
+3.  As a cluster operator, I want the system to be able to tolerate bursts of requests to unidle a
+    service without thrashing
+
+## Design
+
+The requirements for unidling an idle service S with replication controller R are as follows:
+
+1.  The first request to S after it has become idle must:
+    1.  trigger unidling of S
+    2.  be buffered (timeout-bounded) until a destination pod is available
```

Comment:
Okay, based on reading further, I think you intend this to be only HTTP.  It would be good to call that out.  

Does that assumption then prevent this from working with TLS?  Should support TCP-layer buffering?  If so, then what is the action in case of timeout?  Do nothing and have the client retransmit?   If that works for TLS, then maybe use it for HTTP also?   Just thinking out loud here, not sure.